### PR TITLE
uORB: Subscription add copy/move constructor and assignment

### PR DESCRIPTION
**Pull request by dagar**
The only part of the code base that uses this is logger when the LoggerSubscription is copied into the allocated subscription array.

### CODE REVIEW

File: src/modules/uORB/Subscription.hpp [hunk 1]
```diff
@@ -84,6 +84,38 @@ class Subscription
...
+		_orb_id = other._orb_id;
+		_instance = other._instance;
+	}
+
+	// Move constructor
+	Subscription(const Subscription &&other) noexcept
+	{
+		_orb_id = other._orb_id;
+		_instance = other._instance;
+	};
```
Comment (bkueng) [inline src/modules/uORB/Subscription.hpp:99]:
```suggestion
	}
```

File: src/modules/uORB/Subscription.hpp [hunk 2]
```diff
@@ -84,6 +84,38 @@ class Subscription
 		subscribe();
 	}
 
+	// Copy constructor
+	Subscription(const Subscription &other)
```
Comment (bkueng) [inline src/modules/uORB/Subscription.hpp:88]: You can directly initialize them:
```suggestion
	Subscription(const Subscription &other) : _orb_id(other._orb_id), _instance(other._instance)
```

File: src/modules/uORB/Subscription.hpp [hunk 3]
```diff
@@ -84,6 +84,38 @@ class Subscription
...
+	}
+
+	// move assignment
+	Subscription &operator=(Subscription &&other) noexcept
+	{
+		unsubscribe();
+		_orb_id = other._orb_id;
+		_instance = other._instance;
+		return *this;
+	};
```
Comment (bkueng) [inline src/modules/uORB/Subscription.hpp:117]:
```suggestion
	}
```

**Review by bkueng — DISMISSED** [review]

File: src/modules/uORB/Subscription.hpp (hunk 1 above)
Comment (dagar) [reply src/modules/uORB/Subscription.hpp:99]: Fixed.

File: src/modules/uORB/Subscription.hpp (hunk 2 above)
Comment (dagar) [reply src/modules/uORB/Subscription.hpp:88]: Thanks, done.

File: src/modules/uORB/Subscription.hpp (hunk 3 above)
Comment (dagar) [reply src/modules/uORB/Subscription.hpp:117]: Fixed.

**Pull request merged**